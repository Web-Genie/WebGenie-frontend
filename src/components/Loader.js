import React, { useEffect, useState } from "react";
import DotLoader from "react-spinners/DotLoader";
import styled from "styled-components";

import useLogout from "../hooks/useLogout";
function Loader() {
  const [, setCurrentTimer] = useState(0);
  const { handleLogout } = useLogout();

  useEffect(() => {
    const intervalId = setInterval(() => {
      setCurrentTimer((currentCounter) => {
        if (currentCounter === 8) {
          handleLogout();
        }

        return (currentCounter += 1);
      });
    }, 1000);

    return () => {
      clearInterval(intervalId);
    };
  }, []);

  return (
    <LoaderContainer>
      <DotLoader color="#023E8A" size={60} />
    </LoaderContainer>
  );
}

const LoaderContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
  width: 100%;
`;

export default Loader;
