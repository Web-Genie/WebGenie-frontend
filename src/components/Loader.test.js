import renderer from "react-test-renderer";

import Loader from "./Loader";

it("renders correctly", () => {
  const loadingAnimation = renderer.create(<Loader />);

  expect(loadingAnimation).toMatchSnapshot();
});
