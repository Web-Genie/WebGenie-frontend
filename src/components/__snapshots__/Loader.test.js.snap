// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bczRLJ forqeM"
>
  <span
    className="css-1kttl6"
  >
    <span
      className="css-1rj2xb9"
    />
    <span
      className="css-1h57f6d"
    />
  </span>
</div>
`;
